### PR TITLE
Admins can invite users.

### PR DESCRIPTION
**Pull request by robertsapunarich**
This is nowhere near ready, but since we're using a new gem, I want to get feedback. Check out the `devise_invitable` documentation [here](https://github.com/scambra/devise_invitable). 

Add devise_invitable to Gemfile.
Create devise_invitable_add_to_users migration.
Run migration to update db schema.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -14,4 +14,8 @@ def check_if_first_user
     return true
   end
```
Comment (kenmazaika) [inline app/models/user.rb:18]: Does the gem require you implement this method?  If not, I think it might not be used.

**Comment by kenmazaika** [comment]
Made one note, but over all it looks good so far!

File: test/controllers/users/invitations_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require 'test_helper'
+
+class Users::InvitationsControllerTest < ActionController::TestCase
+  test "admin can access invite page" do    
+    user = User.create(:email => "example@example.com", :password => "password", :password_confirmation => "password", :admin => true)
+    sign_in user
+    get :new
```
Comment (robertsapunarich) [inline test/controllers/users/invitations_controller_test.rb:12]: This line returns the following error message whenever I run `rake`:

`NoMethodError: undefined method`name' for nil:NilClass`

Comment (kenmazaika) [inline test/controllers/users/invitations_controller_test.rb:12]: Can you post the full stacktrace?

Comment (robertsapunarich) [inline test/controllers/users/invitations_controller_test.rb:12]: That would probably be helpful. Here it is:

1) Error:
Users::InvitationsControllerTest#test_admin_can_access_invite_page:
NoMethodError: undefined method `name' for nil:NilClass
    test/controllers/users/invitations_controller_test.rb:7:in`block in class:InvitationsControllerTest'

Comment (kenmazaika) [inline test/controllers/users/invitations_controller_test.rb:12]: Is this controller provided courtesy of the `devise_invitable` gem?  If it is, we can go ahead and assume the code works (in general you don't want to spend time testing things that gems provide).

**Comment by robertsapunarich** [comment]
Good point. The controller is provided by the gem. I'll remove the test.

**Comment by Matt-Lepley** [comment]
Looks like there are some merge conflicts, my guess is either in your controller tests or the schema file, or both. Other than that this looks solid.

On a side note, I had no idea of this gem, nice find.

**Comment by robertsapunarich** [comment]
Resolved the merge conflicts. Let me know what you all think.

File: test/controllers/users/invitations_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  test "admin can access invite page" do
+    user = User.create(:email => "example1@example.com", :password => "password", :password_confirmation => "password", :admin => true)
+    sign_in user
+    get :new
+    assert_response :success
+  end
+
+  test "non-admin user cannot access invite page" do
+    user1 = User.create(:email => "example1@example.com", :password => "password", :password_confirmation => "password", :admin => true)
```
Comment (kenmazaika) [inline test/controllers/users/invitations_controller_test.rb]: Small nitpick, but the `user1` isn't really used so we could remove that line.

File: app/controllers/users/invitations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Users::InvitationsController < Devise::InvitationsController
```
Comment (kenmazaika) [inline app/controllers/users/invitations_controller.rb:1]: @amarkpark was talking about refactoring admin stuff under an admin namespace.  You guys might want to sync up to talk about if it makes sense to do this here too.  If we're doing it in one place, we should do this in both places.

**Comment by kenmazaika** [comment]
Looks good to me.  Made one small nitpick but looks like it's ready to go from me. Let  me know what you think and I'll press the merge button.

**Pull request merged**